---
title: Description of Output
description: You've won the trophy!
---
By default, all output files stored into module-specific subdirectories within the `output/` directory. However, each module in the pipeline includes an parameter (`--output_dir`) for the user to define the output directory.

Click on a tab below for details on output for each module ...

import { Tabs, TabItem, Icon } from '@astrojs/starlight/components';

<Tabs>
  <TabItem label="Preprocessing 🔵">

  | Subdirectory | Output Files | Description of Files |
  | --- | --- | --- |
  | `abra` | `*.abra.log` | number of reads before and after realignment for quality; WES only |
  | `alfred` | `*.alfred.qc.[json.gz / summary.txt / txt.gz]` | in-depth quality evaluation genome-wide alignment metrics of final BAM | 
  | `fastqc` | `*_fastqc.[html / zip]` | in-depth quality evaluation on a per base and per sequence manner |
  | `finalBams` | `*.final.[bam / bai]` | final preprocessed BAM and index for downstream analysis |
  | `laneMergedFastqs` | `*.merged.fastq.gz` | concatenated FASTQs; lane split only |
  | `trimmomatic` | `*.trimmomatic.log` | number of reads before and after trimming for quality |

  ---

  :::tip[Provenance]
  Further description of output file types and how to use them ...
  [<Icon name="information" color="goldenrod" size="1.5rem"/>](https://pblaney.github.io/mgp1000-docs/reference/ref/#output)
  :::

  </TabItem>
  <TabItem label="Germline 🟢">

  | Output File | Description of File |
  | --- | --- |
  | `*.deepvariant.germline.snp.indel.[vcf.gz / tbi]` | germline SNP and InDel VCF |
  | `*.fastngsadmix.23.[qopt / log]` | per patient estimated admixture proportions |
  
  ---

  :::tip[Provenance]
  Further description of output file types and how to use them ...
  [<Icon name="information" color="goldenrod" size="1.5rem"/>](https://pblaney.github.io/mgp1000-docs/reference/ref/#output-1)
  :::

  </TabItem>
  <TabItem label="Somatic ⚫️">
  The main output is found in the `somatic/consensus` output directory, where each sample has a subdirectoy with the following files:

  | Output File | Description of File |
  | --- | --- |
  | `*.hq.union.consensus.somatic.snv.txt.gz` | consensus SNV table |
  | `*.hq.union.consensus.somatic.indel.txt.gz` | consensus InDel table |
  | `*.hq.union.consensus.somatic.cnv.bed` | consensus CNV BED |
  | `*.hq.union.consensus.somatic.sv.bedpe` | consensus SV BEDPE |
  | `*.hq.union.consensus.somatic.sv.intersection.plot.pdf` | intersection of SV caller consensus |
  
  ---

  :::tip[Provenance]
  Further description of output file types and how to use them ...
  [<Icon name="information" color="goldenrod" size="1.5rem"/>](https://pblaney.github.io/mgp1000-docs/reference/ref/#output-2)
  :::

  #

  Additional per-tool subdirectories included in the base `output/somatic` output directory:

  | Subdirectory | Description of Files |
  | --- | --- |
  | `battenberg` | native output of Battenberg |
  | `conpair` | native output of Conpair |
  | `delly` | native output of DELLY2 |
  | `facets` | native output of FACETS |
  | `igcaller` | native output of IgCaller |
  | `manta` | native output of Manta |
  | `mutect` | native output of Mutect2 |
  | `sexOfSamples` | sample sex estimation using alleleCount |
  | `strelka` | native output of Strelka2 |
  | `svaba` | native output of SvABA |
  | `telomerecat` | native output of Telomerecat |
  | `telomereHunter` | native output of TelomereHunter |
  | `varscan` | native output of VarScan2 |

  </TabItem>
</Tabs>
