---
title: Description of Output
description: You've won the trophy!
---
By default, all output files stored into module-specific subdirectories within the `output/` directory. However, each module in the pipeline includes an parameter (`--output_dir`) for the user to define the output directory.

Click on a tab below for details on output for each module ...

import { Tabs, TabItem } from '@astrojs/starlight/components';

<Tabs>
  <TabItem label="Preprocessing">

  | Subdirectory | Output Files | Description of Files |
  | --- | --- | --- |
  | `trimLogs` | `*.trim.log` | number of reads before and after trimming for quality |
  | `fastqc` | `*_fastqc.[html / zip]` | in-depth quality evaluation on a per base and per sequence manner | 
  | `alignmentFlagstats` | `*.alignment.flagstat.log` | initial number of reads of various alignment designation |
  | `markdupFlagstats` | `*.markdup.[log / flagstat.log]` | number of detected duplicate reads, number of reads after deduplication |
  | `finalPreprocessedBams` | `*.final.[bam / bai]` | final preprocessed BAM and index for downstream analysis |
  | `coverageMetrics` | `*.coverage.metrics.txt` | genome-wide coverage metrics of final BAM |
  | `gcBiasMetrics` | `*.gcbias.[metrics.txt / metrics.pdf / summary.txt]` | genome-wide GC bias metrics of final BAM |

  </TabItem>
  <TabItem label="Germline">

  | Germline Output File | Description of File |
  | --- | --- |
  | `*.deepvariant.germline.snp.indel.vcf.gz` | germline SNP and InDel VCF |
  | `*.fastngsadmix.23.qopt` | per patient estimated admixture proportions, first two rows with the names of the populations analyzed and the converged upon estimates, and then 100 rows with the bootstrapping estimates |

  </TabItem>
  <TabItem label="Somatic">

  | Somatic Consensus Output File | Description of File |
  | --- | --- |
  | `*.hq.union.consensus.somatic.snv.txt.gz` | consensus SNV table |
  | `*.hq.union.consensus.somatic.indel.txt.gz` | consensus InDel table |
  | `*.hq.union.consensus.somatic.cnv.bed` | consensus CNV BED |
  | `*.hq.union.consensus.somatic.sv.bedpe` | consensus SV BEDPE |
  | `*.hq.union.consensus.somatic.sv.intersection.plot.pdf` | intersection of SV caller consensus |

  Additional per-tool subdirectories included in the base `output/somatic` output directory:

  | Subdirectory | Description of Files |
  | --- | --- |
  | `battenberg` | native output of Battenberg |
  | `conpair` | native output of Conpair |
  | `delly` | native output of DELLY2 |
  | `facets` | native output of FACETS |
  | `igcaller` | native output of IgCaller |
  | `manta` | native output of Manta |
  | `mutect` | native output of Mutect2 |
  | `sexOfSamples` | sample sex estimation using alleleCount |
  | `strelka` | native output of Strelka2 |
  | `svaba` | native output of SvABA |
  | `telomerecat` | native output of Telomerecat |
  | `telomereHunter` | native output of TelomereHunter |
  | `varscan` | native output of VarScan2 |

  </TabItem>
</Tabs>
